cmd/kpod/create.go

```diff
@@ -0,0 +1,899 @@
+	networkAlias   []string          //network-alias
+	nsIPC          string            // ipc
+	nsNet          string            //net
+	nsPID          string            //pid
+	nsUser         string
+	pod            string   //pod
+	privileged     bool     //privileged
+	publish        []string //publish
+	publishAll     bool     //publish-all
+	readOnlyRootfs bool     //read-only
+	resources      createResourceConfig
+	rm             bool              //rm
+	securityOpts   []string          //security-opt
+	sigProxy       bool              //sig-proxy
+	stopSignal     string            // stop-signal
+	stopTimeout    int64             // stop-timeout
+	storageOpts    []string          //storage-opt
+	sysctl         map[string]string //sysctl
+	tmpfs          []string          // tmpfs
+	tty            bool              //tty
+	user           uint32            //user
+	group          uint32            // group
+	volumes        []string          //volume
+	volumesFrom    []string          //volumes-from
+	workDir        string            //workdir
+}
+
+var createDescription = "Creates a new container from the given image or" +
+	" storage and prepares it for running the specified command. The" +
+	" container ID is then printed to stdout. You can then start it at" +
+	" any time with the kpod start <container_id> command. The container" +
+	" will be created with the initial state 'created'."
+
+var createCommand = cli.Command{
+	Name:        "create",
+	Usage:       "create but do not start a container",
+	Description: createDescription,
+	Flags:       createFlags,
+	Action:      createCmd,
+	ArgsUsage:   "IMAGE [COMMAND [ARG...]]",
+}
+
+func createCmd(c *cli.Context) error {
+	// TODO should allow user to create based off a directory on the host not just image
+	// Need CLI support for this
+	if err := validateFlags(c, createFlags); err != nil {
+		return err
+	}
+	//runtime, err := getRuntime(c)
+	runtime, err := libpod.NewRuntime()
+	if err != nil {
+		return errors.Wrapf(err, "error creating libpod runtime")
+	}
+
+	createConfig, err := parseCreateOpts(c, runtime)
+	if err != nil {
+		return err
+	}
+
+	// Deal with the image after all the args have been checked
+	createImage := runtime.NewImage(createConfig.image)
+	if !createImage.HasImageLocal() {
+		// The image wasnt found by the user input'd name or its fqname
+		// Pull the image
+		fmt.Printf("Trying to pull %s...", createImage.PullName)
+		createImage.Pull()
+	}
+
+	runtimeSpec, err := createConfigToOCISpec(createConfig)
+	if err != nil {
+		return err
+	}
+
+	imageName, err := createImage.GetFQName()
+	if err != nil {
+		return err
+	}
+	fmt.Println(imageName)
+	imageID, err := createImage.GetImageID()
+	if err != nil {
+		return err
+	}
+	ctr, err := runtime.NewContainer(runtimeSpec, libpod.WithRootFSFromImage(imageID, imageName, false))
+	if err != nil {
+		return err
+	}
+
+	if err := ctr.Create(); err != nil {
+		return err
+	}
+
+	if c.String("cid-file") != "" {
```

Comment:
Is this required for compatability? Not sure why we're doing it otherwise.

Reply:
It is in the docker cli?  If it makes no sense then we can drop it but document the fact in the man page.